pkg/types/installconfig.go

```diff
@@ -383,3 +386,33 @@ func (c *InstallConfig) WorkerMachinePool() *MachinePool {
 
 	return nil
 }
+
+// ClusterVersionCapabilitiesDefault is a constant representing the
+// default policy for optional, core cluster components
+// +kubebuilder:validation:Enum="";Include;Exclude
+type ClusterVersionCapabilitiesDefault string
+
+const (
+	// IncludeClusterVersionCapabilities indicates that optional,
+	// core cluster components should be included by default
+	IncludeClusterVersionCapabilities = "Include"
+
+	// ExcludeClusterVersionCapabilities indicates that optional,
+	// core cluster components should be excluded by default
+	ExcludeClusterVersionCapabilities = "Exclude"
+)
+
+// Capabilities defines the configuration for user to install/exclude optional OpenShift components
+type Capabilities struct {
+	// InclusionDefault declares the default policy for optional, core cluster components
+	// +optional
+	InclusionDefault ClusterVersionCapabilitiesDefault `json:"inclusionDefault,omitempty"`
+
+	// Include declares capabilities that should be included, regardless of inclusionDefault
+	// +optional
+	Include []string `json:"include,omitempty"`
```

Comment:
Do we want to use a type for the capability names?
```suggestion
	Include []Capability `json:"include,omitempty"`
```